android/tangram/src/com/mapzen/tangram/MapView.java

```diff
@@ -22,13 +28,93 @@ public MapView(Context context, AttributeSet attrs) {
 
     }
 
-    private void configureGLSurfaceView() {
+    /**
+     * Interface for receiving a {@code MapController} once it is ready to be used
+     */
+    public interface OnMapReadyCallback {
+
+        /**
+         * Called when the map is ready to be used
+         * @param mapController A non-null {@code MapController} instance for this {@code MapView}
+         */
+        void onMapReady(MapController mapController);
+
+    }
+
+    /**
+     * Construct a {@code MapController} asynchronously; may only be called from the UI thread
+     * @param callback The object to receive the resulting MapController in a callback;
+     * the callback will be made on the UI thread
+     * @param sceneFilePath Location of the YAML scene file within the asset bundle
+     */
+    public void getMapAsync(final OnMapReadyCallback callback, final String sceneFilePath) {
+
+        final Context context = getContext();
+
+        getMapTask = new AsyncTask<Void, Void, MapController>() {
+
+            @Override
+            protected MapController doInBackground(Void... params) {
+                return new MapController(context, sceneFilePath);
+            }
+
+            @Override
+            protected void onPostExecute(MapController map) {
+                map.setView(glSurfaceView);
```

Comment:
Hm. If any of those are somehow null at this point, there's been some kind of error by the client application in managing the application lifecycle. The correct response to that is to throw an exception, which is what will happen in the methods here. 

We could create our own, slightly more informative exception here to denote that an invalid state has been reached.